mlir/test/fusion/pr-e2e/mixr-gemm/mixr-gemm-broadcast.e2d.mlir

```diff
@@ -0,0 +1,19 @@
+// RUN: rocmlir-driver -kernel-pipeline migraphx,highlevel %s | rocmlir-gen -ph -print-results -rand none - | rocmlir-driver -arch %arch -c  | mlir-cpu-runner -O2 --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_float16_utils%shlibext --entry-point-result=void | FileCheck %s
```

Comment:
Here is an example I d like to see this working : 

```mlir
func.func @mlir_transpose_slice_dot_add_add_tanh_add_sigmoid_sub_mul_mul_add_real(%arg0: !migraphx.shaped<1x64xf32, 64x1>, %arg1: !migraphx.shaped<32x64xf32, 64x1>, %arg2: !migraphx.shaped<32x64xf32, 64x1>, %arg3: !migraphx.shaped<32x64xf32, 64x1>, %arg4: !migraphx.shaped<32x64xf32, 64x1>, %arg5: !migraphx.shaped<32x64xf32, 64x1>, %arg6: !migraphx.shaped<32x64xf32, 64x1>, %arg7: !migraphx.shaped<128x64xf32, 64x1>) -> !migraphx.shaped<32x64xf32, 64x1> attributes {kernel = "mixr", num_cu = 48 : i64} {
    %0 = migraphx.multibroadcast %arg0 {out_dyn_dims = [], out_lens = [32, 64]} : <1x64xf32, 64x1> -> <32x64xf32, 0x1>
    %1 = migraphx.transpose %arg7 {permutation = [1, 0]} : <128x64xf32, 64x1> -> <64x128xf32, 128x1>
    %2 = migraphx.slice %1 {axes = [1], ends = [128], starts = [64]} : <64x128xf32, 128x1> -> <64x64xf32, 64x1>
    %3 = migraphx.dot %arg6, %2 : <32x64xf32, 64x1>, <64x64xf32, 64x1> -> <32x64xf32, 64x1>
    %4 = migraphx.add %3, %0 : <32x64xf32, 64x1>, <32x64xf32, 0x1> -> <32x64xf32, 64x1>
    %5 = migraphx.add %arg1, %4 : <32x64xf32, 64x1>, <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %6 = migraphx.tanh %5 : <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %7 = migraphx.add %arg2, %arg3 : <32x64xf32, 64x1>, <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %8 = migraphx.sigmoid %7 : <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %9 = migraphx.sub %arg4, %8 : <32x64xf32, 64x1>, <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %10 = migraphx.mul %9, %6 : <32x64xf32, 64x1>, <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %11 = migraphx.mul %8, %arg5 : <32x64xf32, 64x1>, <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    %12 = migraphx.add %10, %11 : <32x64xf32, 64x1>, <32x64xf32, 64x1> -> <32x64xf32, 64x1>
    return %12 : !migraphx.shaped<32x64xf32, 64x1>
  }
```

Reply:
@manupak Can you clarify how the optimization we're discussing applies to that example? I don't see it here.

Reply:
Yea -- it was not related to the optimization but just wanted to highlight this is also a foldable broadcast.

an example could be -- also that would also work with your logic (the traversal you mentioned earlier)

128 x f32
 |
V
4 x 32 x f32
 |
V
16 x 4 x 32 x f32

Reply:
ok Im just realizing there is no batch dimension at all. sigh. nvm

Reply:
anyways, my ask was to add something that does not have 'just'  a multibroadcast.

Reply:
After having a second thought, this is actually a counter-example where there is a broadcast on the first dimension where we should *NOT* fold.